File mupen64_src-0.5/compare_pipe is a fifo while file mupen64_src-0.5.fixed/compare_pipe is a fifo
diff -u -r mupen64_src-0.5/main/gui_gtk/main_gtk.c mupen64_src-0.5.fixed/main/gui_gtk/main_gtk.c
--- mupen64_src-0.5/main/gui_gtk/main_gtk.c	2005-08-26 15:08:13.000000000 -0500
+++ mupen64_src-0.5.fixed/main/gui_gtk/main_gtk.c	2005-09-09 23:00:17.000000000 -0500
@@ -1881,7 +1881,7 @@
 	     unsigned char byte;
 	     dest = fopen(temp, "wb");
 	     strcpy(orig, WITH_HOME);
-	     strcat(orig, "share/mupen64/mupen64.ini");
+	     strcat(orig, "lib/mupen64/mupen64.ini");
 	     src = fopen(orig, "rb");
 	     while(fread(&byte, 1, 1, src))
 	       fwrite(&byte, 1, 1, dest);
@@ -1893,7 +1893,7 @@
 	strcpy(temp, g_WorkingDir);
 	strcat(temp, "lang");
 	strcpy(orig, WITH_HOME);
-	strcat(orig, "share/mupen64/lang");
+	strcat(orig, "lib/mupen64/lang");
 	symlink(orig, temp);
 	
 	/*strcpy(temp, g_WorkingDir);
@@ -1906,14 +1906,14 @@
 	strcat(temp, "plugins");
 	mkdir(temp, 0700);
 	strcpy(orig, WITH_HOME);
-	strcat(orig, "share/mupen64/plugins");
+	strcat(orig, "lib/mupen64/plugins");
 	dir = opendir(orig);
 	while((entry = readdir(dir)) != NULL)
 	  {
 	     if(strcmp(entry->d_name + strlen(entry->d_name) - 3, ".so"))
 	       {
 		  strcpy(orig, WITH_HOME);
-		  strcat(orig, "share/mupen64/plugins/");
+		  strcat(orig, "lib/mupen64/plugins/");
 		  strcat(orig, entry->d_name);
 		  src = fopen(orig, "rb");
 		  if(src == NULL) continue;
@@ -1939,7 +1939,7 @@
 		  strcat(temp, "plugins/");
 		  strcat(temp, entry->d_name);
 		  strcpy(orig, WITH_HOME);
-		  strcat(orig, "share/mupen64/plugins/");
+		  strcat(orig, "lib/mupen64/plugins/");
 		  strcat(orig, entry->d_name);
 		  symlink(orig, temp);
 	       }
diff -u -r mupen64_src-0.5/main/main.c mupen64_src-0.5.fixed/main/main.c
--- mupen64_src-0.5/main/main.c	2005-08-26 14:52:11.000000000 -0500
+++ mupen64_src-0.5.fixed/main/main.c	2005-09-09 22:06:12.000000000 -0500
@@ -37,6 +37,8 @@
 
 #include <stdlib.h>
 #include <unistd.h>
+#include <dirent.h>
+#include <sys/stat.h>
 
 #include "main.h"
 #include "guifuncs.h"
@@ -263,7 +265,7 @@
 	     unsigned char byte;
 	     dest = fopen(temp, "wb");
 	     strcpy(orig, WITH_HOME);
-	     strcat(orig, "share/mupen64/mupen64.ini");
+	     strcat(orig, "lib/mupen64/mupen64.ini");
 	     src = fopen(orig, "rb");
 	     while(fread(&byte, 1, 1, src))
 	       fwrite(&byte, 1, 1, dest);
@@ -275,7 +277,7 @@
 	strcpy(temp, g_WorkingDir);
 	strcat(temp, "lang");
 	strcpy(orig, WITH_HOME);
-	strcat(orig, "share/mupen64/lang");
+	strcat(orig, "lib/mupen64/lang");
 	symlink(orig, temp);
 	
 	/*strcpy(temp, g_WorkingDir);
@@ -288,14 +290,14 @@
 	strcat(temp, "plugins");
 	mkdir(temp, 0700);
 	strcpy(orig, WITH_HOME);
-	strcat(orig, "share/mupen64/plugins");
+	strcat(orig, "lib/mupen64/plugins");
 	dir = opendir(orig);
 	while((entry = readdir(dir)) != NULL)
 	  {
 	     if(strcmp(entry->d_name + strlen(entry->d_name) - 3, ".so"))
 	       {
 		  strcpy(orig, WITH_HOME);
-		  strcat(orig, "share/mupen64/plugins/");
+		  strcat(orig, "lib/mupen64/plugins/");
 		  strcat(orig, entry->d_name);
 		  src = fopen(orig, "rb");
 		  if(src == NULL) continue;
@@ -321,7 +323,7 @@
 		  strcat(temp, "plugins/");
 		  strcat(temp, entry->d_name);
 		  strcpy(orig, WITH_HOME);
-		  strcat(orig, "share/mupen64/plugins/");
+		  strcat(orig, "lib/mupen64/plugins/");
 		  strcat(orig, entry->d_name);
 		  symlink(orig, temp);
 	       }
diff -u -r mupen64_src-0.5/Makefile mupen64_src-0.5.fixed/Makefile
--- mupen64_src-0.5/Makefile	2005-08-26 14:52:13.000000000 -0500
+++ mupen64_src-0.5.fixed/Makefile	2005-09-09 22:45:40.000000000 -0500
@@ -166,7 +166,7 @@
 endif
 
 PREFIX		=$(shell grep WITH_HOME config.h | cut -d '"' -f 2)
-SHARE		="$(PREFIX)share/mupen64/"
+SHARE		="$(PREFIX)lib/mupen64/"
 
 PLUGINS		=plugins/mupen64_input.so plugins/blight_input.so plugins/mupen64_hle_rsp_azimer.so plugins/dummyaudio.so plugins/mupen64_audio.so plugins/jttl_audio.so plugins/mupen64_soft_gfx.so plugins/glN64.so
 
@@ -405,7 +405,7 @@
 	
 clean:
 	find . -name '*.o' -print0 | xargs -0r rm -f
-	rm mupen64 mupen64_nogui mupen64_dbg plugins/mupen64_input.so blight_input/arial.ttf.c blight_input/ttftoh plugins/blight_input.so plugins/mupen64_hle_rsp_azimer.so plugins/dummyaudio.so plugins/mupen64_audio.so plugins/jttl_audio.so plugins/mupen64_soft_gfx.so plugins/glN64.so
+	rm -f mupen64 mupen64_nogui mupen64_dbg plugins/mupen64_input.so blight_input/arial.ttf.c blight_input/ttftoh plugins/blight_input.so plugins/mupen64_hle_rsp_azimer.so plugins/dummyaudio.so plugins/mupen64_audio.so plugins/jttl_audio.so plugins/mupen64_soft_gfx.so plugins/glN64.so
 
 clean_o:
 	find . -name '*.o' -print0 | xargs -0r rm -f
